src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -92,7 +94,35 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             // No AT in the cache or AT needs to be refreshed
             try
             {
-                return await RefreshRtOrFetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                if (msalAccessTokenItem == null)
+                {
+                    return await RefreshRtOrFetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    if (msalAccessTokenItem.NeedsRefresh())
+                    {
+                        _ = Task.Run(async () =>

```

Comment:
this code block is duplicated 3 times (except RefreshRtOrFetchNewAccessTokenAsync)